OCPCLOUD-1782: e2e workflow for vSphere upi cluster with the External platform type

lobziik opened the pull request:
Upi with the `External` platform config steps and related workflows.

The config step is based on `upi-conf-vsphere` and mainly replicates it. Unfortunately, I didn't invent how to hook and patch generated manifests without extensive changes within `upi-conf-vsphere`, so, decided to place `External` platform related things separately.

Requires MCO update https://github.com/openshift/machine-config-operator/pull/3567 to be merged to progress with worker nodes spinning.

- https://github.com/openshift/machine-config-operator/pull/3567

lobziik commented:
/pj-rehearse

lobziik commented:
/pj-rehearse

elmiko commented:
cc @rvanderp3

lobziik commented:
this need to be merged for further progress: https://github.com/openshift/machine-config-operator/pull/3567

lobziik commented:
/cc @rvanderp3 @jcpowermac

rvanderp3 commented on ci-operator/step-registry/upi/conf/vsphere/platform-external/upi-conf-vsphere-platform-external-commands.sh:
```diff
@@ -0,0 +1,228 @@
+rm -f openshift/99_openshift-cluster-api_worker-machineset-*.yaml
+
+### Update platform spec in the infra resource
+echo "Patching infrastructure resource manifest with the 'External' platform type..."
+cat <<EOF > /tmp/infraPlatformSpecPatch.yml
+spec:
+  platformSpec:
+    type: External
+    external:
+      platformName: vSphere
+status:
+  platform: External
+  platformStatus:
+      type: External
+      external: {}
+EOF
+yq-go m -x -i "manifests/cluster-infrastructure-02-config.yml" "/tmp/infraPlatformSpecPatch.yml"
```
could just the manifests be modified in a step following upi-conf-vsphere instead of replicating the bulk of the step?  i'm not necessarily opposed to replicating if we need to.

lobziik replied:
This will require splitting `upi-conf-vsphere` step and passing all manifests between steps.
Now, `upi-conf-vsphere` generates manifests and uses them instantly for generating ignition files, manifests get deleted then. 

I will try to do so, but not sure how many jobs it might affect.

rvanderp3 replied:
no worries @lobziik , i was just curious

rvanderp3 reviewed (commented):
only found one nit, otherwise looks good

rvanderp3 commented:
/lgtm

lobziik commented:
/retest

lobziik commented:
/pj-rehearse

lobziik commented:
/test pj-rehearse

lobziik commented:
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.13-provider-certification-tool-vsphere-upi-platform-external

lobziik commented:
I removed the usage of `openshift-e2e` test step since there is no platform-agnostic version for now.
Instead of that, I added a job that leverages the certification tool. Added it next to the existing certification-tool ci.

/cc @mtulio @elmiko PTAL

lobziik commented:
/pj-rehearse

elmiko reviewed (commented):
this is generally looking good to me

rvanderp3 commented:
looks good to me :)

/lgtm

mtulio commented:
I will left some comments with regards the notifications:
/hold

mtulio commented on ci-operator/jobs/redhat-openshift-ecosystem/provider-certification-tool/redhat-openshift-ecosystem-provider-certification-tool-main-periodics.yaml:
```diff
@@ -80,6 +80,81 @@ periodics:
+    ci.openshift.io/generator: prowgen
+    job-release: "4.13"
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.13-provider-certification-tool-vsphere-upi-platform-external
```
Can we enable the Slack notifications as we used to do in the other [two jobs](https://github.com/openshift/release/blob/021658c71bb2eb8783c78f259ac0a13f4e79f7e1/ci-operator/jobs/redhat-openshift-ecosystem/provider-certification-tool/redhat-openshift-ecosystem-provider-certification-tool-main-periodics.yaml#L174-L180)?

lobziik replied:
Done. I also altered report templates for other jobs to contain links to a specific run.

PTAL https://github.com/openshift/release/pull/36572/commits/bfeb4b050f9ba4fd00a5b4945f2f8c1635886264

mtulio reviewed (commented):
Looks good to me. You can unhold it after setting the notifications. :)
/lgtm

mtulio commented:
/lgtm ++

mtulio commented:
/lgtm

lobziik commented:
/hold cancel

lobziik commented:
The previous run finished successfully, no need to rerun, since nothing critical was changed in the last commit.

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/36572/rehearse-36572-periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.13-provider-certification-tool-vsphere-upi-platform-external/1630941489901080576

/pj-rehearse ack
